### PR TITLE
[Caffe2] Fix the legacy padding handling on global pool case

### PR DESCRIPTION
**Pull request by houseroad**
Add the missing logic back.

### CODE REVIEW

**Comment by houseroad** [comment]
The missed logic is here: https://github.com/caffe2/caffe2/commit/bc2ecc84a775eaea72a237c4292b66cefe6218af#diff-8bedfaa8298a968cddebba1f40126a43L242

File: caffe2/onnx/onnx_exporter.cc [hunk 1]
```diff
@@ -282,16 +282,19 @@ ConvertedResult OnnxExporter::CreateConvPoolNodes(
     CAFFE_ENFORCE(
         node.op_type().size() >= 4 &&
         (node.op_type().rfind("Pool") == node.op_type().size() - 4));
-    CAFFE_ENFORCE(!global);
     const auto& input_size = shapes.at(node.input(0));
     const auto& output_size = shapes.at(node.output(0));
     CAFFE_ENFORCE(output_size.dims().size() == 4);
-    if (legacy_pad_attr.i() ==
+    if (node.op_type().find("Global") == 0) {
```
Comment (bddppq) [inline caffe2/onnx/onnx_exporter.cc]: check `global` variable instead?

File: caffe2/onnx/onnx_exporter.cc [hunk 2]
```diff
@@ -282,16 +282,19 @@ ConvertedResult OnnxExporter::CreateConvPoolNodes(
...
+      // No special handling, since global pool does not care about the pad.
+    } else if (legacy_pad_attr.i() ==
+        static_cast<int64_t>(caffe2::LegacyPadding::NOTSET)) {
```
Comment (bddppq) [inline caffe2/onnx/onnx_exporter.cc]: Remove the last clause at line 321

Comment (houseroad) [reply caffe2/onnx/onnx_exporter.cc]: good catch

File: caffe2/onnx/onnx_exporter.cc [hunk 3]
```diff
@@ -282,16 +282,19 @@ ConvertedResult OnnxExporter::CreateConvPoolNodes(
     CAFFE_ENFORCE(
         node.op_type().size() >= 4 &&
         (node.op_type().rfind("Pool") == node.op_type().size() - 4));
-    CAFFE_ENFORCE(!global);
     const auto& input_size = shapes.at(node.input(0));
     const auto& output_size = shapes.at(node.output(0));
     CAFFE_ENFORCE(output_size.dims().size() == 4);
-    if (legacy_pad_attr.i() ==
+    if (global) {
+      // No special handling, since global pool does not care about the pad.
```
Comment (yinghai) [inline caffe2/onnx/onnx_exporter.cc]: This empty struct looks a bit weird. Can we do `if(!global) { if else }`?

**Comment by bddppq** [comment]
@onnxbot retest this please

**Review by bddppq — APPROVED** [review]

**Pull request merged**